clang/cmake/modules/AddClang.cmake

```diff
@@ -114,7 +114,7 @@ macro(add_clang_library name)
     if(TARGET "obj.${name}")
       target_compile_definitions("obj.${name}" PUBLIC CLANG_BUILD_STATIC)
     endif()
-  elseif(NOT ARG_SHARED AND NOT ARG_STATIC)
+  elseif(NOT ARG_SHARED AND NOT ARG_STATIC AND NOT ARG_CLANG_IMPORT)
```

Comment:
This seems like it would change the behaviour of other libraries which were previously being built for shared linking (or at least were exporting their interfaces) to now be built for static linking (or no longer export their symbols).

Reply:
what do you mean, its only checking for the arg not being set, so it shouldn't change the behaviour for other libraries.

Reply:
Then we should be using `if (NOT DEFINED ...)`. The syntax you have can check for the truthiness of the value.

Reply:
`cmake_parse_args` always defines option values:

> For the <options> keywords, these variables will always be defined, and they will be set to TRUE if the keyword is present, or FALSE if it is not.

see: https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html

So `ARG_CLANG_IMPORT` will always be defined, it may be defined either to `True` or `False`.